src/lib/analytics/AnalyticsClass.js

```diff
@@ -150,22 +150,24 @@ export class AnalyticsClass {
 
     //fire all events on  GA also
     if (googleAnalytics) {
-      this.fireGoogleAnalyticsEvent(event, data)
+      const gaEvent = this.convertToGA(event, data)
+      this.fireGoogleAnalyticsEvent(ANALYTICS_EVENT, gaEvent)
     }
 
     logger.debug('fired event', { event, data })
   }
 
-  // convertToGA = (data: any = {}) => {
-  //   const values = Object.values(data)
-  //   const eventValues = remove(values, x => typeof x === 'number') //remove returns the removed items, so eventValues will be numbers
-  //   const eventStrings = remove(values, x => typeof x === 'string')
-  //   const gaEvent = {
-  //     eventValue: eventValues.shift(),
-  //     eventLabel: eventStrings.shift() || eventValues.shift() || JSON.stringify(values.shift()),
-  //   }
-  //   return gaEvent
-  // }
+  convertToGA = (event: string, data: any = {}) => {
+    const eventValue = find(data, isNumber) // get first numberic value
+    const eventString = find(data, isString) // get first string value
+    const firstValue = first(values(data)) // get first value as a fallback
```

Comment:
@serdiukov-o-nordwhale @MarianoAmado 
this is exactly why I keep asking to stop refactoring. no your refactoring does not have same results as previous logic.
consider that data has 1 numeric value and one object value.
in your code both eventValue and eventLabel will get the first numeric value.
while in my code eventValue gets the numeric value and eventLabel the object value.

Reply:
> so i chose to get the first numeric value (ie shift()) and put it in eventValue
> and i chose to take the first string and put it in eventLabel

first number = value
first string = label

Reply:
@sirpy fixed, if no strings, we're falling back to the first **object** value (JSON-stringified)